compat/inet_ntop.cc

```diff
@@ -204,8 +203,8 @@ inet_ntop6(const u_char *src, char *dst, size_t size)
     /*
      * Format the result.
      */
-    tp = tmp;
```

Comment:
You have (correctly) moved `tp` declaration to this formatting code where `tp` is used. Why not move the `tmp` declaration as well? The two are closely related and used in the same code.

Reply:
Sure; it won't change scope end as it's used at the very end of the function but this won't hurt.
I'm also renaming it as "buf" because in the end it's a buffer.